webapp/passport/js/MxN/protein.js

```diff
@@ -192,188 +74,212 @@ protein =
+            return '++';
+        }
+        else if (charge === 3) {
+            return '+++';
         }
-        return cvSum / count;
+        else if (charge === -1) {
+            return '-';
+        }
+        else if (charge === -2) {
+            return '--';
+        }
+        else if (charge === -3) {
+            return '---';
+        }
+        return (charge >= 0 ? '+' : '') + charge;
     },
 
     initialize: function(data) {
 
         protein.projects = proteinJSON.projects;
         protein.features = proteinJSON.features;
         protein.sequence = proteinJSON.sequence;
+        protein.preferredname = proteinJSON.preferredname;
 
-        protein.peptides = [];
         protein.precursors = [];
 
-        var peptideGrouped = LABKEY.vis.groupData(data.rows, function (row) {
-            return row.PeptideId;
-        });
+        // Abbreviate the peptide sequences like Skyline
+        data.rows.forEach(function(row) {
+            row.dataType = 'Peptide';
+            row.fragment = row.ModifiedSequence;
+        }, this);
+        LABKEY.targetedms.QCPlotLegendHelper.setupLegendPrefixes(data.rows, 3);
 
-        for (var peptideId in peptideGrouped) {
-            var peptideRows = peptideGrouped[peptideId];
-
-
-            protein.peptides.push(peptide);
-        }
-
-        var precursorGrouped = LABKEY.vis.groupData(data.rows, function (row) {
+        const precursorGrouped = LABKEY.vis.groupData(data.rows, function (row) {
             return row.PrecursorId;
         });
 
-        for (var precursorId in precursorGrouped) {
-
-            var precursorRows = precursorGrouped[precursorId];
-
-            var charge = precursorRows[0].Charge;
-            var mz = precursorRows[0].Mz;
-
-            var precursor = {
-                sequence: precursorRows[0].PeptideSequence + ' ' + (charge >= 0 ? '+' : '') + charge + ' ' + mz.toFixed(2),
+        Object.keys(precursorGrouped).forEach(function(precursorId ) {
+            const precursorRows = precursorGrouped[precursorId];
+            const precursor = precursorRows[0];
+
+            const precursorData = {
+                Sequence: precursor.PeptideSequence,
+                FullDescription: precursor.ModifiedSequence + ' ' + protein.renderCharge(precursor.Charge) + ', ' + precursor.Mz.toFixed(4),
+                TotalArea: 0,
+                StartIndex: precursor.StartIndex,
+                EndIndex: precursor.EndIndex,
+                Enabled: true,
+                PrecursorId: precursor.PrecursorId,
+                ChromatogramId: precursor.PrecursorChromInfoId,
+                PeptideId: precursor.PeptideId,
+                CalibrationCurveId: precursor.CalibrationCurveId,
+
+                peptideSequence: precursor.PeptideSequence,
+                modifiedSequence: precursor.ModifiedSequence,
+                mz: precursor.Mz,
+                peptideId: precursor.PeptideId,
+                precursorId: precursor.PrecursorId,
```

Comment:
there is PrecursorId present at L181

Reply:
Thanks for the reminder. I collapsed two different data structures into one and failed to clean up the duplicates and casing. Now consistent and de-duped